great_expectations/profile/basic_dataset_profiler.py

```diff
@@ -313,24 +325,34 @@ def _create_expectations_for_numeric_column(cls, dataset, column):
         dataset.expect_column_median_to_be_between(column, min_value=value - 1, max_value=value + 1)
 
-        result = dataset.expect_column_quantile_values_to_be_between(column,
-                                                                     quantile_ranges={
-                                                                         "quantiles": [0.05, 0.25, 0.5, 0.75, 0.95],
-                                                                         "value_ranges": [[None, None], [None, None],
-                                                                                          [None, None], [None, None],
-                                                                                          [None, None]]
-                                                                     },
-                                                                     result_format="SUMMARY"
-                                                                     )
-        dataset.expect_column_quantile_values_to_be_between(column,
-                                                            quantile_ranges={
-                                                                "quantiles": result.result["observed_value"][
-                                                                    "quantiles"],
-                                                                "value_ranges": [[v - 1, v + 1] for v in
-                                                                                 result.result["observed_value"][
-                                                                                     "values"]]
-                                                            }
-                                                            )
+        try:
+            # TODO these are not implemented correctly on sqlite, and likely other sql dialects
+            result = dataset.expect_column_quantile_values_to_be_between(
+                column,
+                quantile_ranges={
+                    "quantiles": [0.05, 0.25, 0.5, 0.75, 0.95],
+                    "value_ranges": [
+                        [None, None],
+                        [None, None],
+                        [None, None],
+                        [None, None],
+                        [None, None],
+                    ],
+                },
+                result_format="SUMMARY",
+            )
+            dataset.expect_column_quantile_values_to_be_between(
+                column,
+                quantile_ranges={
+                    "quantiles": result.result["observed_value"]["quantiles"],
+                    "value_ranges": [
+                        [v - 1, v + 1] for v in
+                        result.result["observed_value"]["values"]
+                    ],
+                },
+            )
+        except OperationalError as e:
+            logger.warning(e)
```

Comment:
Instead of a try/except block here, it seems like we should be setting `catch_exceptions` to true on the expectation evaluation itself, and then deciding what to do next using that information; otherwise we end up recreating the logic of the try/catch embedded in the validate method.

That means there would be an if on the result value set on line 320 based on whether an exception was raised.

Lastly: I know this doesn't originate in this PR, but as I look at this code, I also think that rather than evaluate the expectation twice in a row, the second one here could be declared without evaluation (turn off `interactive_evaluation` right before the declaration) for a substantial speedup.

Reply:
Excellent suggestions, both!